it/controllers/pensions/paymentsIntoOverseasPensions/PensionsCustomerReferenceNumberControllerISpec.scala

```diff
@@ -0,0 +1,264 @@
+
+        import Selectors._
+        import user.commonExpectedResults._
+
+        "render the customer reference number page with correct content and no pre-filling" which {
+          implicit lazy val result: WSResponse = showPage(user, anPensionsUserDataEmptyCya)
+
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY))
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraph1, paragraphSelector(1))
+          textOnPageCheck(hintText, hintTextSelector)
+          inputFieldValueCheck(inputName, inputSelector, "")
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(pensionCustomerReferenceNumberUrl(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render the page with correct content with pre-filling" which {
+          val pensionsCustomerReferenceNumber = "PENSIONSINCOME245"
+
+          val pensionsViewModel = aPaymentsIntoOverseasPensionsViewModel.copy(
+            customerReferenceNumberQuestion = Some(pensionsCustomerReferenceNumber)
+          )
+          val pensionUserData = pensionUserDataWithOverseasPensions(pensionsViewModel)
+          implicit lazy val result: WSResponse = showPage(user, pensionUserData)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY))
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraph1, paragraphSelector(1))
+          textOnPageCheck(hintText, hintTextSelector)
+          inputFieldValueCheck(inputName, inputSelector, pensionsCustomerReferenceNumber)
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(pensionCustomerReferenceNumberUrl(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+      }
+    }
+
+    "Redirect to the pension summary page if there is no session data" should {
+      lazy val result: WSResponse = getResponseNoSessionData
+
+      "has an SEE_OTHER status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(pensionSummaryUrl(taxYearEOY))
+      }
+    }
+  }
+
+  ".submit" should {
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+
+        s"return $BAD_REQUEST error when incorrect format is submitted" which {
```

Comment:
Perhaps it should read something like "... when an empty value is submitted".

Reply:
Test case name updated

Reply:
Confirmed.